quipucords/api/insights_report/serializers.py

```diff
@@ -0,0 +1,92 @@
+# Copyright (C) 2022  Red Hat, Inc.
+
+# This software is licensed to you under the GNU General Public License,
+# version 3 (GPLv3). There is NO WARRANTY for this software, express or
+# implied, including the implied warranties of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv3
+# along with this software; if not, see
+# https://www.gnu.org/licenses/gpl-3.0.txt.
+
+"""Insights report serializers."""
+
+from rest_framework import fields
+from rest_framework.serializers import Serializer, ValidationError
+
+from api.insights_report.constants import CANONICAL_FACTS
+
+default_kwargs = dict(required=False)
+
+
+class SystemProfileSerializer(Serializer):
+    number_of_cpus = fields.IntegerField(source="cpu_count", **default_kwargs)
+    system_memory_bytes = fields.IntegerField(
+        **default_kwargs
+    )  # TODO: not mapped yet to a fingerprint
+    infrastructure_type = fields.CharField(max_length=100, **default_kwargs)
+    infrastructure_vendor = fields.CharField(
+        max_length=100, **default_kwargs
+    )  # TODO: Looks like virt_type might answer this, but only for virtualized infra
+    # yupana builds operating_system from os_release
+    os_release = fields.CharField(max_length=100, **default_kwargs)
+    arch = fields.CharField(source="architecture", max_length=50, **default_kwargs)
+
+
+class YupanaHostSerializer(Serializer):
+    display_name = fields.CharField(source="name", **default_kwargs)
+    bios_uuid = fields.CharField(**default_kwargs)
+    fqdn = fields.CharField(
+        source="name", **default_kwargs
+    )  # TODO: not sure on this one, just followed old implementation https://github.com/quipucords/quipucords/blob/da644172fb35ad7128ba20372cf64fbf7ff4f367/quipucords/fingerprinter/task.py#L373
+    insights_id = fields.CharField(source="insights_client_id", **default_kwargs)
+    ip_addresses = fields.ListField(child=fields.CharField(), **default_kwargs)
+    mac_addresses = fields.ListField(child=fields.CharField(), **default_kwargs)
+    provider_id = fields.CharField(
+        source="name", **default_kwargs
+    )  # TODO: not sure on this one
+    provider_type = fields.CharField(source="cloud_provider", **default_kwargs)
+    satellite_id = fields.CharField(
+        **default_kwargs
+    )  # TODO: not sure on this one https://github.com/RedHatInsights/insights-host-inventory/blob/813a290f3a1c702312d8e02d1e59ba328c6f8143/swagger/api.spec.yaml#L901-L907
+    subscription_manager_id = fields.CharField(**default_kwargs)
+    system_profile = SystemProfileSerializer(source="*", **default_kwargs)
```

Comment:
`source=*` means that the same data available to the "parent" serializer (`YupanaHostSerializer`) will be passed over to the "child" `system_profile`.

The entity both will use is essentially the `SystemFingerprints` model with some extra steps

Reply:
SystemFingerprint model: https://github.com/quipucords/quipucords/blob/c603b2f9d73906c1ff32b95fa4ec1faa3d8cf707/quipucords/api/deployments_report/model.py#L68-L185